android/src/main/java/com/reactnativecommunity/webview/RNCWebView.java

```diff
@@ -387,20 +404,33 @@ public ThemedReactContext getThemedReactContext() {
   }
 
   protected class RNCWebViewBridge {
+        private String TAG = "RNCWebViewBridge";
         RNCWebView mWebView;
+        String injectedObjectJson;
 
         RNCWebViewBridge(RNCWebView c) {
           mWebView = c;
         }
 
+        public void setInjectedObject(String s) {
+            injectedObjectJson = s;
+        }
+
         /**
          * This method is called whenever JavaScript running within the web view calls:
          * - window[JAVASCRIPT_INTERFACE].postMessage
          */
         @JavascriptInterface
         public void postMessage(String message) {
-          mWebView.onMessage(message);
+            if (mWebView.getMessagingEnabled()) {
+                mWebView.onMessage(message);
+            } else {
+                FLog.w(TAG, "ReactNativeWebView.postMessage method was called but messaging is disabled. Pass an onMessage handler to the WebView.");
+            }
         }
+
+        @JavascriptInterface
+        public String injectedObject() { return injectedObjectJson; }
```

Comment:
I would vote for the getter/setter to also use the `Json` suffix (especially the getter, since that's what's exposed to the Javascript in the WebView, which needs to know that this is JSON to be parsed?)

Reply:
nit: I also wonder if it would be useful for this to use the `get` prefix, just to make clear when consuming in the Javascript side that it's a function, not a property? I don't feel strongly about that though

Reply:
`window.ReactNativeWebView.injectedObjectJson()` is already pretty long, so I think I'll skip the `get` prefix unless you feel really strongly about it.